clang/lib/Analysis/LifetimeSafety/LifetimeAnnotations.cpp

```diff
@@ -71,6 +71,88 @@ bool implicitObjectParamIsLifetimeBound(const FunctionDecl *FD) {
   return isNormalAssignmentOperator(FD);
 }
 
+// Decl::isInStdNamespace will return false for iterators in some STL
+// implementations due to them being defined in a namespace outside of the std
+// namespace.
+static bool isInStlNamespace(const Decl *D) {
+  const DeclContext *DC = D->getDeclContext();
+  if (!DC)
+    return false;
+  if (const auto *ND = dyn_cast<NamespaceDecl>(DC))
+    if (const IdentifierInfo *II = ND->getIdentifier()) {
+      StringRef Name = II->getName();
+      if (Name.size() >= 2 && Name.front() == '_' &&
+          (Name[1] == '_' || isUppercase(Name[1])))
+        return true;
```

Comment:
probably an off-topic comment: the code looks suspicious to me, if a namespace name with the prefix `_[_A-Z]`, we consider it is a STL namespace. the code was introduced in https://reviews.llvm.org/D66152, @Xazax-hun do you still remember the rationale?